### ISSUE TITLE
error in run time, Something severely wrong happened!!! All the pixels are invalid in estimation Laplacian noise!!!

### ISSUE THREAD

**Issue by HymEric** [alterzero/RBPN-PyTorch#3]
(pytorch100) hym@vipa210:~/pycharm_remote_files/RBPN-pytorch$ CUDA_VISIBLE_DEVICES=3 python eval.py
Namespace(chop_forward=True, data_dir='./Vid4', file_list='foliage.txt', future_frame=True, gpu_mode=True, gpus=1, model='weights/RBPN_4x.pth', model_type='RBPN', nFrames=7, other_dataset=True, output='Results/', residual=False, seed=123, testBatchSize=1, threads=1, upscale_factor=4)
===> Loading datasets
===> Building model  RBPN
/temp_disk/hym/pycharm_remote_files/RBPN-pytorch/dbpns.py:50: UserWarning: nn.init.kaiming_normal is now deprecated in favor of nn.init.kaiming_normal_.
  torch.nn.init.kaiming_normal(m.weight)
/temp_disk/hym/pycharm_remote_files/RBPN-pytorch/dbpns.py:54: UserWarning: nn.init.kaiming_normal is now deprecated in favor of nn.init.kaiming_normal_.
  torch.nn.init.kaiming_normal(m.weight)
/temp_disk/hym/pycharm_remote_files/RBPN-pytorch/rbpn.py:74: UserWarning: nn.init.kaiming_normal is now deprecated in favor of nn.init.kaiming_normal_.
  torch.nn.init.kaiming_normal(m.weight)
/temp_disk/hym/pycharm_remote_files/RBPN-pytorch/rbpn.py:78: UserWarning: nn.init.kaiming_normal is now deprecated in favor of nn.init.kaiming_normal_.
  torch.nn.init.kaiming_normal(m.weight)
neigbor frame- is not exist
neigbor frame- is not exist
neigbor frame- is not exist
Constructing pyramid...done!
Pyramid level 6All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!

Pyramid level 5All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!
All the pixels are invalid in estimation Laplacian noise!!!
Something severely wrong happened!!!

**Comment by alterzero** [comment]
This error messages are from pyflow.
I think you might not compile it properly.

**Comment by HymEric** [comment]
Thank you very much. When I compile pyfow, run python eval.py. The results .png in Results/Vid4/foliage_4x is not like a picture exactly. All of results as below:
![1_RBPNF7_看图王](https://user-images.githubusercontent.com/26487821/55923072-83c18d00-5c36-11e9-86c3-0702ec475249.png)

And this is the main running log:
![image](https://user-images.githubusercontent.com/26487821/55923307-807ad100-5c37-11e9-8d9e-a75f1176bdb7.png)
![M){ 2UM_YOC(MO8RY_VZ%SD](https://user-images.githubusercontent.com/26487821/55923318-912b4700-5c37-11e9-9d37-ace4e8ce6919.png)

Do you know is there a problem?

**Comment by alterzero** [comment]
Looks like you fail to load the weight. Make sure file .pth is available with the same naming.
Then, I just updated several files as well. Please update yours.
Thank you.

**Comment by HymEric** [comment]
Thanks s lot. I will try.

**Comment by HymEric** [comment]
I have a question, What the meaning of **"neigbor frame- is not exist".** in runing time. Is it normal?

**Comment by alterzero** [comment]
"neigbor frame- is not exist" -> eg. 1st frame does not have previous frame, it will show this message. 

computation time -> by mistake I set chop_forward = True (as default setting), you can change this to False if you have enough GPU memory.

**Comment by DouYishun** [comment]
Hi @HymEric , after I re-compile the pyflow and pass the pyflow/demo.py, there still exist above pyflow error "All the pixels are invalid in estimation Laplacian noise!!!".
How did you solve it?

**Comment by alterzero** [comment]
For the pyflow problem, try to use python 3.5, I run my code on python 3.5, it should work.
For the second issue, it was by intention I made that, because as you mentioned it is inevitable.

**Comment by DouYishun** [comment]
Thanks for your reply.
I test the provided model RBPN_4x.pth on Vid4.foliage with following configuration
![image](https://user-images.githubusercontent.com/15257616/56797593-9ac3ca00-6847-11e9-9a93-49a335ee0cf7.png)
and I get PSNR=24.77, which is lower than 26.27 in Table 7.
(I have removed first and last two frames, cropped 8-pixels near image boundary.)

Does the provided model trained with 6-PF?
What is the expected performance of the RBPN_4x.pth on Vid4.foliage?

**Comment by alterzero** [comment]
The calculation of PSNR is on luminance channel only, not RGB

**Comment by mangdian** [comment]
@HymEric Hi, I meet the same problem. When using pyflow, it always gives "Something severely wrong happened". Would you please tell me how to fix it?

**Comment by Cugtyt** [comment]
Same problem

**Comment by Entretoize** [comment]
No solution from 2019 ?
